In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

# Set visual style
sns.set_theme(style="whitegrid")

# Load the cleaned dataset
df = pd.read_csv('cleaned_titanic.csv')

# Display basic information
print(f"Dataset Shape: {df.shape}")
df.head()

Dataset Shape: (891, 11)


,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,S
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,S
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,S


In [3]:
# Statistical Summary of Numerical Features
print("=== Numerical Summary ===")
display(df.describe().T)

# Survival Rate Breakdown
print("\n=== Survival Rate by Gender ===")
display(df.groupby('Sex')['Survived'].agg(['count', 'mean']))

print("\n=== Survival Rate by Passenger Class ===")
display(df.groupby('Pclass')['Survived'].agg(['count', 'mean']))

=== Numerical Summary ===


,count,mean,std,min,25%,50%,75%,max
PassengerId,891.0,446.000000,257.353842,1.00,223.5000,446.0000,668.5,891.0000
Survived,891.0,0.383838,0.486592,0.00,0.0000,0.0000,1.0,1.0000
Pclass,891.0,2.308642,0.836071,1.00,2.0000,3.0000,3.0,3.0000
Age,891.0,29.112424,13.304424,0.42,21.5000,26.0000,36.0,80.0000
SibSp,891.0,0.523008,1.102743,0.00,0.0000,0.0000,1.0,8.0000
Parch,891.0,0.381594,0.806057,0.00,0.0000,0.0000,0.0,6.0000
Fare,891.0,32.204208,49.693429,0.00,7.9104,14.4542,31.0,512.3292



=== Survival Rate by Gender ===


,count,mean
Sex,,
female,314,0.742038
male,577,0.188908



=== Survival Rate by Passenger Class ===


,count,mean
Pclass,,
1,216,0.629630
2,184,0.472826
3,491,0.242363


## Summary Statistics Overview

- **Overall Survival Rate:** 38.38% (342 out of 891 passengers survived).
- **Average Age:** ~29.1 years (ranging from infants under 1 year to 80 years old).
- **Average Fare:** $32.20 (median fare is $14.45, indicating right-skewed high values).
- **Gender Distribution:** Female passengers had a ~74.2% survival rate compared to ~18.9% for male passengers.

In [ ]:
fig, axes = plt.subplots(2, 2, figsize=(14, 10))

# Plot 1: Survival Rate by Class and Sex
sns.barplot(data=df, x='Pclass', y='Survived', hue='Sex', palette='Set2', ax=axes[0, 0])
axes[0, 0].set_title('1. Survival Rate by Class & Sex', fontsize=12, fontweight='bold')
axes[0, 0].set_ylabel('Survival Rate')

# Plot 2: Age Distribution by Survival
sns.kdeplot(data=df, x='Age', hue='Survived', common_norm=False, fill=True, palette='coolwarm', ax=axes[0, 1])
axes[0, 1].set_title('2. Age Distribution Density by Survival Status', fontsize=12, fontweight='bold')

# Plot 3: Survival Rate by Family Size
df['FamilySize'] = df['SibSp'] + df['Parch'] + 1
sns.barplot(data=df, x='FamilySize', y='Survived', palette='Blues_d', ax=axes[1, 0])
axes[1, 0].set_title('3. Survival Rate by Family Size', fontsize=12, fontweight='bold')
axes[1, 0].set_ylabel('Survival Rate')

# Plot 4: Log Fare Distribution by Survival
sns.boxplot(data=df, x='Survived', y='Fare', palette='Set3', ax=axes[1, 1])
axes[1, 1].set_yscale('log')
axes[1, 1].set_title('4. Fare Distribution (Log Scale)', fontsize=12, fontweight='bold')

plt.tight_layout()
plt.show()

## Key Exploratory Insights & Findings

1. **Gender Priority ("Women First"):**
   Female passengers survived at a significantly higher rate (**74.20%**) than male passengers (**18.89%**).
2. **Socio-Economic Privilege:**
   Class status strongly dictated survival probability. First-class passengers achieved a **62.96%** survival rate, compared to **24.24%** for third-class passengers.
3. **Child Evacuation Trend:**
   Children aged 12 and under experienced higher survival rates (**57.97%**) relative to working-age adults.
4. **Family Size Dynamics:**
   Solo travelers had lower survival rates (**30.35%**). Small families (2–4 members) fared best (peaking at **72.41%** for families of 4), while large families (5+ members) faced steep drops in survival (< 20%).
5. **Fare Anomalies & High Variance:**
   The fare column spans from **$0.00** (15 complimentary/crew records) to extreme outliers of **$512.33**, demonstrating significant economic disparity aboard the vessel.